torch/fx/experimental/symbolic_shapes.py

```diff
@@ -411,6 +411,32 @@ def is_concrete_bool(a: Union[bool, SymBool]) -> bool:
 def is_nested_int(s):
     return isinstance(s, torch.SymInt) and s.node.is_nested_int()
 
+# helper function for constructing a symbolic nested int from a nested int
+def _create_symbolic_nested_int(nested_int, base_source, shape_env):
+    # Base source is None in two cases:
+    # (1) tensor._base is _dummy_instance OR
+    # (2) tensor is an intermediate
+    from torch._dynamo.source import EphemeralSource
+
+    # check if the nested int is already symbolic
+    if is_symbolic(nested_int):
+        if nested_int.node.shape_env is shape_env:
+            return nested_int
+        else:
+            nested_int = nested_int.node.hint
+
+    nested_source: Source = EphemeralSource("intermediate_offsets_or_lengths")
```

Comment:
Nice! Always using EphemeralSource sounds like a good idea for now to avoid having to thinking about saving source on Tensor until we actually need to.

Reply:
exactly! booted that til later for the best chance of getting something landable now. none of the test cases we currently have try to guard on a nested int, so we don't need a non-ephemeral source